### PR TITLE
Feature/accepttomemorypoolworker

### PR DESCRIPTION
**Pull request by lodenos**

### CODE REVIEW

**Comment by nkostoulas** [comment]
The test looks exactly the same with the whitelisting one. How does it test freezelist/burnlist/redemption??

File: src/policy/policy.cpp [hunk 1]
```diff
@@ -113,48 +113,44 @@ bool IsBurn(const CTransaction& tx)
...
-  for (uint32_t itr = 0; itr < tx.vout.size(); ++itr) {
+  for (uint32_t itrA = 0; itrA < tx.vout.size(); ++itrA) {
     vector<vector<uint8_t>> vSolutions;
-    if (Solver(tx.vout[itr].scriptPubKey, whichType, vSolutions))
-      if (whichType == TX_PUBKEYHASH && uint160(vSolutions[0]).IsNull()) {
-        if (vSolutions.size() < 2)
-          return false;
-        for (uint32_t itr = 1; itr < vSolutions.size(); ++itr) {
-          CKeyID keyId = CKeyID(uint160(vSolutions[0]));
+    if (Solver(tx.vout[itrA].scriptPubKey, whichType, vSolutions))
```
Comment (nkostoulas) [inline src/policy/policy.cpp]: wasn't this correct before the change? @tomt1664 ?

More than one vout, where the first vout is a null output. I think that's why your tests pass. They should be failing when IsRedemption() is checked.

Comment (tomt1664) [reply src/policy/policy.cpp]: The rule is: if a transaction has a P2PKH output with a null(0) address, then all other P2PKH outputs must be on the  addressFreezelist. (non-P2PKH outputs can only be TX_FEE or TX_NULL_OUTPUT as enforced by IsWhitelisted). 

So if tx has no null addresses: pass. If a null address, all others must be checked. 

The new function is correct (except I think itrB should start at index 0: I'll check, but I think the tx outputs are marked as frozen if the null address is in any vout position).

Comment (nkostoulas) [reply src/policy/policy.cpp]: Ahh okay. How is this null address generated? Is it the 0,0 point on the curve?

**Comment by tomt1664** [comment]
The test does not need to test whitelist functions, and the freezelist and burnlist will not have all the addresses on the whitelist. 

I think an adequate test for these two functions would do the following:

1. Have freezelist and burnlist enabled. 
2. With nothing added to either list, send a redemption transactions (i.e. will null address) and check that it gets blocked. 
3. Send a transaction that has one or more OP_NULL_TX outputs and check that it gets blocked. 
4. Create a redemption transaction where the first output is to a null(0) address, and the second output is to a wallet address. 
5. Add the address to the freezelist (via RPC) and then send the transaction, and check it is accepted. 
6. Create a burn transaction (sending everything to an OP_NULL_TX output). 
7. Add the same address to the burnlist and check that the burn transaction is accepted.

**Pull request merged**